### DISCUSSION TITLE
Error: Unsuccessful HTTP response

### DISCUSSION THREAD

**Discussion by Jamtjim** [iTwin/itwinjs-core#2613] [Q&A] [answered]
As a competitive newbie to iTwin.js, I watched the developer program videos (https://www.youtube.com/playlist?list=PL6YCKeNfXXd_dXq4u9vtSFfsP3OTVcL8N) and it seemed simple enough. So I followed all the instructions, installed all the necessary packages and appended the code so that it pointed to my demo model. However when I tried to test, I was greeted by the following error message:

![image](https://user-images.githubusercontent.com/93394507/139461952-b70c942d-650e-476a-987a-1047e10b6fa9.png)

Can anyone tell me where I might be going wrong?

**Comment by ColinKerr** [comment]
It seems like you have an authorization error (because you got a 401 response).  It's not clear to me why you got this error, you can find more info on setting an application here: 
[Web Application Quick Start](https://developer.bentley.com/tutorials/web-application-quick-start/)
[Register and Modify an Application](https://developer.bentley.com/tutorials/register-and-modify-application/)

It this doesn't help please provide some more detailed information on the code you have now where you are in the lesson plan.

**Comment by BaoTon** [comment, marked as answer]
Unsuccessful HTTP response usually stems from an invalid authority.

You'll need to make sure this variable exists in your .env file:
IMJS_AUTH_AUTHORITY="https://ims.bentley.com"

If you're following the youtube series, this environment variable does not yet exist since the video is a bit outdated. 
If you've ran the CRA template and somehow removed the variable, please make sure to add this variable again.

**Reply by BaoTon** [reply]
An updated Chapter 3 Lesson 3 video is on the way that should contain the new CRA template changes and the above mentioned "IMJS_AUTH_AUTHORITY" error.
